### ISSUE TITLE
Pre-generating color data to asset meta

### ISSUE THREAD

**Issue by adevade** [aryehraber/statamic-color-extractor#12]
Is there a way to pre-generate the colors to all assets? Say via a command or something?

This would be a good feature otherwise. Helps speed up the first page load for pages with many images! 👍

**Comment by aryehraber** [comment]
Hey @adevade, this is a great idea and would definitely be useful since, as you said, initial load times can be quite long when using many images. 

I don't have much time to work on such a feature at the moment, but I'd be happy to review a PR! 

Thanks!

**Comment by adevade** [comment]
Nice! I think I could give the PR it a shot! 

Do you think it should be its own command (like `please assets:extract-color-meta`), or should i hook into `please assets:meta`?
That command basically [loops through and saves](https://github.com/statamic/cms/blob/5a9519a4a7a365bd1760c1a9eec6f901c157e659/src/Console/Commands/AssetsMeta.php#L24-L27) all assets, [generating meta files](https://github.com/statamic/cms/blob/5a9519a4a7a365bd1760c1a9eec6f901c157e659/src/Assets/AssetRepository.php#L117-L136).

We could also hook into the `AssetSaved` event, and generate the colors whenever an asset is saved?

EDIT: Or once on `AssetUploaded`? Maybe available as a config option? It could generate all three types at upload.

**Comment by aryehraber** [comment]
Interesting ideas! 

I think a dedicated command like `please assets:extract-color-meta` sounds good, however I'm a little hesitant to generate it for ALL image files, since this can easily run into the hundreds of MBs or even GBs on larger projects, which will take ages. At risk of making it too complex, perhaps the command can take a Container and/or Folder path into account, allowing the user to scope it to just the images they need. Thoughts?

I think auto-generating on `AssetSaved`/`AssetUploaded` would be a nice one (via a config option, of course).

**Comment by Burans** [comment]
Hey! Did this ever get implemented in some way?
I replaced a bunch of images manually in the assets folder and now i need to recalculate colors for all of them.
Thanks!

**Comment by aryehraber** [comment]
@Burans No, nothing was ever implemented, unfortunately. Depending on your needs, maybe you can mass-delete the existing color meta data from the asset meta data files? That way it should get recalculated next time the image is used with the addon.

**Comment by aryehraber** [comment]
For anyone still in need of this; I finally added a [Command](https://github.com/aryehraber/statamic-color-extractor#extract-command) to pre-generate colour data in bulk, and an [AssetUploaded](https://github.com/aryehraber/statamic-color-extractor#auto-extract-on-asset-upload) event listener that auto-generates with each asset upload so that first page visits are never effected 🚀
